.github/workflows/build.yaml

```
@@ -0,0 +1,57 @@
+name: Build Docker image and push to Dockerhub
+
+on:
+  push:
+    branches:
+      - "main"
+env:
+  IMAGE_NAME: stringerrss/stringer
+
+jobs:
+  buildx:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Extract Docker sha tag
+        id: get-tag-sha
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.IMAGE_NAME }}
+          tags: type=sha
+
+      - name: Extract Docker latest tag
+        id: get-tag-latest
+        uses: docker/metadata-action@v4
+        with:
+          images: ${{ env.IMAGE_NAME }}
+          tags: type=raw, value=latest
+
+      - name: Log in to Docker Hub
+        id: login
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build and Push
+        id: build-and-push
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: ./Dockerfile
+          platforms: linux/amd64,linux/arm64
+          push: true
```

I think we might be able to build on *every* Github push and only push to Docker only on `main` with something like:

```suggestion
          push: ${{ github.ref_name == 'main' }}
```